src/reducers/index.js

```diff
@@ -91,9 +92,15 @@ export const reducer = (state = initialState, action) => {
     case 'SET_ROLES':
       return {
         ...state,
-        roles: action.payload,
+        roles: action.payload.roles,
```

Comment:
This line should be changed back to `roles: action.payload` the array itself is being passed as the payload so there's no need to treat it as an object. This is currently causing the front-end to crash.